sympy/printing/latex.py

```diff
@@ -397,7 +400,13 @@ def convert(expr):
                 if self.order not in ('old', 'none'):
                     args = expr.as_ordered_factors()
                 else:
-                    args = expr.args
+                    args = list(expr.args)
+
+                # If angular quantity is present append them at the back
+                for i in range(len(args)):
+                    if str(args[i]) in angular_quantity:
```

Comment:
Don't rely on the `str` printer here, too.

Reply:
If we use `args[i].name` instead of `str(args[i])` we get error when `args[i]` is Integer,Pow etc. Is there any other way so that we don't have to use the `str` printer and we can make this check.